README.md

```diff
@@ -7,17 +7,47 @@ localhost.
 small Cassandra cluster on a local box. It is meant for testing a Cassandra cluster.
 
+New to Python development?
+--------------------------
+Python has moved on since CCM started development. `pip` is the new `easy_install`,
+Python 3 is the new 2.7, and virtualenv is strongly recommended for managing multiple Python versions
+and application dependencies.
+
+A typical MacOS setup would be to install [Homebrew](https://docs.brew.sh/Installation),
+then `brew install python3` to get some version of Python 3 then use virtualenv to
+manage the dependencies for CCM. Make sure to add [brew's bin directory to your path in
+your ~/.zshenv](https://www.zerotohero.dev/zshell-startup-files/). This would be
+`/usr/local` for MacOS Intel and `/opt/homebrew/` for MacOS on Apple Silicon.
+
+To create the virtualenv run `python3 -m venv --prompt ccm venv` with your git repo as the
+current working directory to create an environment for CCM. Then `source venv/bin/activate` to
+enable the venv for the current terminal and `deactivate` to exit.
+
+Now you a ready to set up the venv with CCM and its test dependencies installed. `pip install -e <path_to_ccm_repo>`
+to install CCM, and its runtime dependencies from `requirements.txt`, so that the version of
+CCM you are running points to the code you are actively working on. There is no build or package step because you
+are editing the Python files being run every time you invoke CCM.
+
+Almost there. Now you just need to add the test dependencies that are not in `requirements.txt`.
+`pip install mock pytest` to finish setting up your dev environment!
+
+Another caveat that has recently appeared Cassandra versions 4.0 and below ship with a version of JNA that isn't
+compatible with Apple Silicon and there are no plans to update JNA on those versions. One work around if you are
+generally building Cassandra from source to use with CCM is to replace the JNA jar in your Maven repo with a [newer
+one](https://search.maven.org/artifact/net.java.dev.jna/jna/5.8.0/jar) that supports Apple Silicon.
+Which you version you need to replace will vary depending on the Cassandra version, but it will normally be in
+`~/.m2/repository/net/java/dev/jna/jna/<someversion>`. You can also replace the library in
+`~/.ccm/repository/<whatever>/lib`.
 
 Requirements
 ------------
 
 - A working python installation (tested to work with python 2.7).
-- pyYAML (http://pyyaml.org/ -- `sudo easy_install pyYaml`)
-- six (https://pypi.org/project/six/ -- `sudo easy_install six`)
+- See `requirements.txt` for runtime requirements
+- `mock` and `pytest` for tests
 - ant (http://ant.apache.org/, on Mac OS X, `brew install ant`)
-- psutil (https://pypi.org/project/psutil/)
 - Java (which version depends on the version of Cassandra you plan to use. If
-  unsure, use Java 7 as it is known to work with current versions of Cassandra).
+  unsure, use Java 8 as it is known to work with current versions of Cassandra).
```

Comment:
Perhaps reference Java 11 instead here.  Java 8 is on LTS.

Reply:
Good suggestion. I updated the text. Please LMK what you think.

Reply:
I think maybe 8 makes sense until we still maintain as a Community Cassandra versions like 3.x which are on Java 8. I believe we maintain them until next year or next major release.

Reply:
This comment thread is on an outdated version of the diff. I updated it to reflect that we build on 8 and support 11.

Reply:
Sounds great! Thank you so much!